### PR TITLE
[com_finder] Add submenu

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
Pull Request for Improvement.
#### Summary of Changes

This PR adds the missing submenu for com_finder.
Smart Search is the only component with sidebar that doesn't have the submenu.

image
#### Testing Instructions

Two tests needed:
###### Install
1. Install clean joomla using https://github.com/andrepereiradasilva/joomla-cms/archive/com_finder_submenu.zip package
2. Check the Components -> Smart Search has a new submenu.
###### Update
1. Go to Components -> Joomla update -> Uplaod and install
2. Install my custom package (download from https://github.com/andrepereiradasilva/update.joomla.org/raw/master/submenu-on-update.zip)
3. Check the Components -> Smart Search has a new submenu.
#### Observations
- I don't understand very well the `lft` and `rgt` database fields. Can anyone check those in the sql insert commands?
- Only tested in Mysql. If anyone can test on postgresql would be great.

### CODE REVIEW

**Comment by andrepereiradasilva** [comment]
@mbabker, so, is this PR i don't need to add the SQL to update, since joomla will read the manisfest on update and create the submenus right?

**Comment by mbabker** [comment]
No, the update SQL is needed.  On install and update Joomla will update the `manifest_cache` column of the extensions table for core extensions with the updated data for the XML manifests.  It doesn't actually run individual extension updates.

File: installation/sql/mysql/joomla.sql [hunk 1]
```diff
@@ -1274,6 +1274,9 @@ INSERT INTO `#__menu` (`id`, `menutype`, `title`, `alias`, `note`, `path`, `link
 (19, 'menu', 'com_joomlaupdate', 'Joomla! Update', '', 'Joomla! Update', 'index.php?option=com_joomlaupdate', 'component', 1, 1, 1, 28, 0, '0000-00-00 00:00:00', 0, 0, 'class:joomlaupdate', 0, '', 35, 36, 0, '*', 1),
 (20, 'main', 'com_tags', 'Tags', '', 'Tags', 'index.php?option=com_tags', 'component', 0, 1, 1, 29, 0, '0000-00-00 00:00:00', 0, 1, 'class:tags', 0, '', 37, 38, 0, '', 1),
 (21, 'main', 'com_postinstall', 'Post-installation messages', '', 'Post-installation messages', 'index.php?option=com_postinstall', 'component', 0, 1, 1, 32, 0, '0000-00-00 00:00:00', 0, 1, 'class:postinstall', 0, '', 39, 40, 0, '*', 1),
+(22, 'menu', 'com_finder_indexed_content', 'Indexed Content', '', 'Smart Search/Indexed Content', 'index.php?option=com_finder', 'component', 0, 18, 2, 27, 0, '0000-00-00 00:00:00', 0, 0, 'class:finder', 0, '', 43, 44, 0, '*', 1),
```
Comment (mbabker) [inline installation/sql/mysql/joomla.sql]: You need to fix the `lft`/`rgt` values for the whole table.  As is it's breaking the nested set structure.

Don't worry about it so much for updates.  It's a long standing open issue that inserting anything via update SQLs into tables with nested sets breaks it.  Would be so nice to get that part of updates moved over to pure PHP but I don't see it happening anytime soon.

Comment (andrepereiradasilva) [inline installation/sql/mysql/joomla.sql]: i have to understand those lft/rgt values ...

BTW, two things:
1. i saw com_finder and their plugins are not in com_admin script.php. This a bug, right?
See https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_admin/script.php#L212-L240

2. Why does the sampel data reinstall all menu items (and not just the new ones), is this just incorrect, or does have some propose?

Comment (mbabker) [inline installation/sql/mysql/joomla.sql]: > i saw com_finder and their plugins are not in com_admin script.php. This a bug, right?

The component's not there, the plugins are (scroll down a few more lines).  It's either oversight or someone removed them, either way all the core extensions should be in that massive array.

> Why does the sample data reinstall all menu items (and not just the new ones), is this just incorrect, or does have some propose?

Ask whoever wrote how the sample data is installed.  It's a massive wreck is all I can say on the subject right now.

Comment (andrepereiradasilva) [inline installation/sql/mysql/joomla.sql]: > The component's not there, the plugins are (scroll down a few more lines). It's either oversight or someone removed them, either way all the core extensions should be in that massive array.

Created a PR to add com_finder https://github.com/joomla/joomla-cms/pull/10300

> Ask whoever wrote how the sample data is installed. It's a massive wreck is all I can say on the subject right now.

Yeah. i was thinking the same...

Thanks again for your help.

Comment (brianteeman) [inline installation/sql/mysql/joomla.sql]: Not at my PC to check as it is 3am but i suspect you will need to add this
to Hathor template

**Comment by andrepereiradasilva** [comment]
i need to understand the lft/rgt values and have a way to rebuild then on update ...
I will put this in standby for now.

**Comment by brianteeman** [comment]
I can help you on this. I will try to submit a pull request to your repo but if I forget please remind me<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10290">issues.joomla.org/joomla-cms/10290</a>.</sub>

**Comment by andrepereiradasilva** [comment]
ok thanks!

**Comment by conconnl** [comment]
@andrepereiradasilva & @brianteeman little reminder :smile: 
This is a good UX improvement, so I would like to test it went completely finished.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10290">issues.joomla.org/joomla-cms/10290</a>.</sub>

**Comment by andrepereiradasilva** [comment]
waitng for brian to have time on this one

**Comment by brianteeman** [comment]
Sorry @andrepereiradasilva what is it I am supposed to be helping with - I am forgetful<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10290">issues.joomla.org/joomla-cms/10290</a>.</sub>

**Comment by andrepereiradasilva** [comment]
andrepereiradasilva said
> i need to understand the lft/rgt values and have a way to rebuild then on update ...
> I will put this in standby for now.

brian said
> I can help you on this. I will try to submit a pull request to your repo but if I forget please remind me

**Comment by brianteeman** [comment]
Ok I will try to remember how

**Comment by brianteeman** [comment]
@andrepereiradasilva 
Michael is right it doesnt really matter for updated and seems to be a bit of a mess anyway

Obviously you need to rename your update files now with the new version ;)

Best thing I can suggest to get the correct lft/rgt values is to runs the sql you have right now. then go to the menu manager. make sure all menu items is selected and click on rebuild

that should give you the correct values

File: installation/sql/mysql/joomla.sql (hunk 1 above)
Comment (brianteeman) [inline installation/sql/mysql/joomla.sql]: ignore the comment about hathor its fine

**Comment by andrepereiradasilva** [comment]
closing as no intenttion to pursue this anymore

**Pull request closed without merging**